src/backend/api/Fusion.Resources.Domain/Commands/Departments/GetRelatedDepartments.cs

```diff
@@ -61,19 +52,38 @@ public Handler(ILogger<Handler> logger, IMediator mediator, ILineOrgResolver lin
             private async Task<QueryRelatedDepartments?> ResolveRelevantDepartmentsAsync(string fullDepartmentPath)
             {
                 var relevantDepartments = new QueryRelatedDepartments();
-                var department = await lineOrgResolver.ResolveDepartmentAsync(DepartmentId.FromFullPath(fullDepartmentPath));
-                if (department is null) return null;
 
-                var children = await lineOrgResolver.ResolveDepartmentChildrenAsync(department);
-                if (children is not null)
-                {
-                    relevantDepartments.Children = await children.ToQueryDepartment(profileService);
-                }
 
-                var siblings = await lineOrgResolver.ResolveDepartmentChildrenAsync(string.Join(" ", fullDepartmentPath.Split(" ").SkipLast(1)));
-                if(siblings is not null)
+                var orgUnit = await lineOrg.ResolveOrgUnitAsync(fullDepartmentPath, a => a.ExpandChildren());
+
+                if (orgUnit is null)
+                    return null;
+
+                // Add children
+
+                // Children should be returned
+                if (relevantDepartments.Children is null)
```

Comment:
Should the relevantDepartment always have children? Even if the relevantdepartment we check is at the "bottom"?

Reply:
If this is set to null it indicates that the server did not include it in it's response, which would either be an error on their site or we have some issue internally where a cached copy is used which has not been fetched correctly..